### ISSUE TITLE
`unique` implementation struggles with overloads

### ISSUE THREAD

**Issue by rgommers** [serge-sans-paille/pythran#1842]
```python
import numpy as np

#pythran export try_unique(float[:])
def try_unique(x):
    return np.unique(x, return_counts=False)
```

A simple test shows the number of return values is wrong:
```python
>>> x = np.ones(7, dtype=float)
>>> try_unique(x)
(array([1.]), array([0]), array([0, 0, 0, 0, 0, 0, 0]), array([7]))
```

This is in a similar boat as the `keepdims` keyword (see gh-1564, gh-1820): a boolean keyword that controls return type. I'd say `unique` is worse actually.

This is going to trip people up, it may be worth considering to support boolean keywords by making separate types for `bool_true` and `bool_false` that can get different overloads. If this is too hard, I think it makes sense to only support the single argument version of `unique`.

**Comment by charlotte12l** [comment]
The returned value is the same as `np.unique(x, return_index=True, return_inverse=True, return_counts=True)`.

```
>>> np.unique(x, return_index=True, return_inverse=True, return_counts=True)
(array([1.]), array([0], dtype=int64), array([0, 0, 0, 0, 0, 0, 0], dtype=int64), array([7], dtype=int64))
```

Looking at the implementation, we'll find for key `return_counts` in `np.unique` , Pythran only supports the following:
```
  template <class E>
  std::tuple<types::ndarray<typename E::dtype, types::pshape<long>>,
             types::ndarray<long, types::pshape<long>>,
             types::ndarray<long, types::pshape<long>>,
             types::ndarray<long, types::pshape<long>>>
  unique(E const &expr, bool return_index, bool return_inverse,
         bool return_counts)
```
, and it will always think the keys are true.

**Comment by serge-sans-paille** [comment]
@charlotte12l before working on the generalization of `numpy.mean` case, you could add a special case for `np.unique`. It's easier to generalize when you have two examples instead of one ;-)

**Comment by serge-sans-paille** [comment]
Fixed by #1876
